pkg/operators/wasm/maps.go

```diff
@@ -236,7 +254,33 @@ func (i *wasmOperatorInstance) mapUpdate(ctx context.Context, m wapi.Module, sta
 		return
 	}
 
-	err = ebpfMap.Update(key, value, ebpfFlags)
+	mapTypesAcceptingFD := []ebpf.MapType{ebpf.ArrayOfMaps, ebpf.HashOfMaps}
+
+	if getFD {
```

Comment:
General question: If we already know when a map expects a FD for the value and return an error when the user doesn't set the flag right:
Why do we have the flag at all? Can't we make this distinction automatic?